examples/event_handler/main.rs

```diff
@@ -0,0 +1,70 @@
+use std::env::var;
+use std::sync::atomic::{AtomicU32, Ordering};
+
+use poise::serenity_prelude as serenity;
+use poise::Event;
+
+// Types used by all command functions
+type Error = Box<dyn std::error::Error + Send + Sync>;
+type Context<'a> = poise::Context<'a, Data, Error>;
+
+// Custom user data passed to all command functions
+pub struct Data {
+    pongs: AtomicU32,
+}
+
+#[tokio::main]
+async fn main() {
+    env_logger::init();
+
+    let options = poise::FrameworkOptions {
+        event_handler: |_ctx, event, _framework, _data| {
+            Box::pin(event_handler(_ctx, event, _framework, _data))
+        },
+        ..Default::default()
+    };
+
+    poise::Framework::builder()
+        .token(
+            var("DISCORD_TOKEN")
+                .expect("Missing `DISCORD_TOKEN` env var, see README for more information."),
+        )
+        .setup(move |_ctx, _ready, _framework| {
+            Box::pin(async move {
+                Ok(Data {
+                    pongs: AtomicU32::new(0),
+                })
+            })
+        })
+        .options(options)
+        .intents(
+            serenity::GatewayIntents::non_privileged() | serenity::GatewayIntents::MESSAGE_CONTENT,
+        )
+        .run()
+        .await
+        .unwrap();
+}
+
+async fn event_handler(
+    ctx: &serenity::Context,
+    event: &Event<'_>,
+    _framework: poise::FrameworkContext<'_, Data, Error>,
+    data: &Data,
+) -> Result<(), Error> {
+    match event {
+        Event::Ready { data_about_bot } => {
+            println!("Logged in as {}", data_about_bot.user.name);
+        }
+        Event::Message { new_message } => {
+            if new_message.content == "!ping" {
```

Comment:
This is a bit of a misleading example because poise has built-in support for text commands, and if someone wanted to implement them manually they should use base serenity instead of poise.

Could you change the code to count, e.g. the number of.. all-uppercase messages, or messages containing a specific keyword, or whatever? I think that'd be a more realistic use case of the event handler

Reply:
Sure, changed it to count Poise.

Reply:
Haha some subtle self advertisement